src/desktop/add-to-path.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2025 Arm Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import * as vscode from 'vscode';
+import { BuiltinToolPath } from './builtin-tool-path';
+import * as os from 'os';
+
+const isWindows = os.platform() === 'win32';
+const PYOCD_BUILTIN_PATH = 'tools/pyocd/';
+
+export function addPyOCDtoPATH(context: vscode.ExtensionContext): void {
+    //get pyOCD path from tools folder
+    const builtinPyocd = new BuiltinToolPath(PYOCD_BUILTIN_PATH);
+    const pathPyOCD = builtinPyocd.getAbsolutePath()?.toString();
+    //get PATH variable
+    const path = process.env.PATH;
+    //slice unwanted characters from pyOCD path, take care of the Windows case
+    let slicedPyOCDpath;
+    if(isWindows) {
+        slicedPyOCDpath = pathPyOCD?.slice(7, pathPyOCD.length - 4);
```

Comment:
Why is that required?

Reply:
Ah, so it is used to capture the absolute path to a specific executable rather than the parent directory, only.

So, perhaps the following code might be a bit cleaner:
```
const PYOCD_BUILTIN_PATH = 'tools/pyocd/pyocd';
const builtinPyocd = new BuiltinToolPath(PYOCD_BUILTIN_PATH);
const pathPyOCD = path.dirname(builtinPyocd.getAbsolutePath()?.fsPath);
```

This code is OS agnostic. We could move the last line down into the `BuiltinToolPath` class:
```
public getAbsoluteDirname(): vscode.Uri | undefined {
    const absUri = this.getAbsolutePath();
    return absUri?.with({ path: path.dirname(absUri?.fsPath) });
}
```

Reply:
We probably want to make the `.exe` extension for Windows optional for the `BuiltinToolPath` class.
Either as an argument to the constructor. Or as an argument to the `getAbsolutePath` method.
Note that it would be absolutely fine to have two `BuiltinToolPath` instances where the one that you need for the environment variable collection is constructed with `tools/pyocd/` rather than `tools/pyocd/pyocd`.
Another option could be to interpret an input string with a trailing slash/backslash as a directory rather than a file.